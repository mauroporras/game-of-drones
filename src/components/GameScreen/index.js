import React from 'react'

import PrimaryTitle from '../PrimaryTitle/'
import SecondaryTitle from '../SecondaryTitle/'
import Footer from '../Footer/'

const GameScreen = props => (
  <div>
    <header>
      <PrimaryTitle text='Game of Drones' />
      <SecondaryTitle text={props.title} />
    </header>
    <main>
      {props.children}
    </main>
    <Footer />
  </div>
)

export default GameScreen
