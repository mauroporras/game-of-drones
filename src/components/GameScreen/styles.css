.GameScreen {
  margin: 0 auto;
  max-width: 35rem;
}
