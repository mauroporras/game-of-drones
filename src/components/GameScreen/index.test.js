import React from 'react'
import renderer from 'react-test-renderer'

import GameScreen from './index'

test("Component renders the correct children", () => {
  const component = renderer.create(
    <GameScreen>
      <div>Game on!</div>
    </GameScreen>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
