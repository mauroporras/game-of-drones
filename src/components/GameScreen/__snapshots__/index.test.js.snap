exports[`test Component renders the correct children 1`] = `
<div
  className="GameScreen">
  <header>
    <h1
      className="PrimaryTitle">
      Game of Drones
    </h1>
    <h2
      className="SecondaryTitle" />
  </header>
  <main>
    <div>
      Game on!
    </div>
  </main>
  <footer
    className="Footer">
    <p
      className="Footer__paragraph">
      Fair play. Have fun!
    </p>
    <a
      className="Footer__link"
      href="https://github.com/mauroporrasp/game-of-drones">
      Github repo
    </a>
  </footer>
</div>
`;
